src/main/java/com/redislabs/redisai/RedisAI.java

```diff
@@ -94,7 +96,55 @@ public boolean setTensor(String key, Object tensor, int[] dimensions){
       throw new RedisAIException(ex);
     }
   }
-  
+
+  /**
+   * TS.GET key
+   *
+   * @param key
+   * @return Tensor
+   */
+  public Tensor getTensor(String key) throws JRedisAIRunTimeException {
+    try (Jedis conn = getConnection()) {
+      List<?> reply = sendCommand(conn, Command.TENSOR_GET, SafeEncoder.encode(key), Keyword.META.getRaw(), Keyword.VALUES.getRaw() ).getObjectMultiBulkReply();
+      if(reply.isEmpty()) {
+        return null;
+      }
+      DataType dtype = null;
+      long[] shape = null;
+      Object values = null;
+      Tensor tensor = null;
+      for (int i = 0; i < reply.size(); i+=2) {
```

Comment:
@gkorland please revise this approach

Reply:
why not switch on String? https://www.geeksforgeeks.org/string-in-switch-case-in-java/

Reply:
:+1: changed in the new commit